File: python/src/hywrapper/models/other/leaderboards_response.py
from __future__ import annotations

from typing import Dict, List

from pydantic import BaseModel, ConfigDict, Field

from hywrapper.models.hypixel_response import HypixelResponse


class Leaderboard(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    path: str = ""
    prefix: str = ""
    title: str = ""
    location: str = ""
    count: int = 0
    leaders: List[str] = Field(default_factory=list)


class LeaderboardsResponse(HypixelResponse):
    leaderboards: Dict[str, List[Leaderboard]] = Field(default_factory=dict)

File: typescript/src/models/other/LeaderboardsResponse.ts
import { HypixelResponse } from "../../types";

export interface LeaderboardsResponse extends HypixelResponse {
    leaderboards?: Record<string, Leaderboard[]>;
}

export interface Leaderboard {
    path?: string;
    prefix?: string;
    title?: string;
    location?: string;
    count?: number;
    leaders?: string[];
}
